tests/test_examples.py

```diff
@@ -10,7 +12,9 @@ def test_examples():
         with open(examples_path / test['test_file']) as datafile:
             data = yaml.safe_load(datafile)
         class_validator = validator[test['definition']]
+        
         try:
-            assert class_validator.validate(data) is None
+            with raises(ValidationError) if test.get("shouldValidationFail") else nullcontext():
+                assert class_validator.validate(data) is None
```

Comment:
```suggestion
            if test.get("shouldValidationFail"):
                with raises(ValidationError):
                    assert class_validator.validate(data) is None
            else:
                assert class_validator.validate(data) is None
```

TIL about `contextlib.nullcontext`! I recommend rewriting this a little more verbosely for later understandability, above is an example of what this might look like. If you were to do so, also consider dropping the contextlib import statement. This is a style preference only.